### PR TITLE
Get the full paths of executables

### PR DESCRIPTION
**Pull request by nam-jaehyun**
- get the full paths of executables (#532)
- add processName and parentProcessName fields (#532)
- fix overwritten flags (#501)

### CODE REVIEW

**Comment by nam-jaehyun** [comment]
the new update would have no issue, but it seems that we have a bcc issue again.

File: KubeArmor/BPF/system_monitor.c [hunk 1]
```diff
@@ -519,51 +525,61 @@ static __always_inline bool prepend_path(struct path *path, bufs_t *string_p)
...
+        struct path *p = exec_map.lookup(&pid_tgid);
+        exec_map.delete(&pid_tgid);
+        return p;
+    } else { // FILE_BUF_TYPE
+        struct path *p = file_map.lookup(&pid_tgid);
+        file_map.delete(&pid_tgid);
+        return p;
+    }
+
+    return NULL;
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
    if (buf_type == EXEC_BUF_TYPE) {
        struct path *p = exec_map.lookup(&pid_tgid);
        exec_map.delete(&pid_tgid);
        return p;
    }
    // FILE_BUF_TYPE
    struct path *p = file_map.lookup(&pid_tgid);
    file_map.delete(&pid_tgid);
    return p;
```

File: KubeArmor/BPF/system_monitor.c [hunk 2]
```diff
@@ -710,6 +726,72 @@ static __always_inline int events_perf_submit(struct pt_regs *ctx)
...
+    bufs_t *bufs_p = get_buffer(DATA_BUF_TYPE);
+    if (bufs_p == NULL)
+        return 0;
+
+    save_context_to_buffer(bufs_p, (void*)&context);
+    save_str_to_buffer(bufs_p, (void *)&string_p->buf[*off]);
+
+    events_perf_submit(ctx);
+
+	return 0;
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
    return 0;
```

Comment (nyrahul) [reply KubeArmor/BPF/system_monitor.c]: needs indentation fixes in multiple places.

File: tests/host_scenarios/kubearmor-dev-next_test_1/cmd1 [hunk 3]
```diff
@@ -1,7 +1,7 @@
 source: kubearmor-dev-next
-cmd: screen -h
+cmd: diff -h
 result: failed
 ---
 operation: Process
-condition: screen
+condition: diff
 action: Block
```
Comment (Ankurk99) [inline tests/host_scenarios/kubearmor-dev-next_test_1/cmd1]: Can you please add a new line at the end of file?

Comment (nam-jaehyun) [reply tests/host_scenarios/kubearmor-dev-next_test_1/cmd1]: sure.

**Comment by nyrahul** [comment]
PR looks good to me. Thanks for the quick turn-around. Lets get this merged asap so that we can make relevant changes on dependent modules (auto-policy needs to use this processPath).

**Comment by Ankurk99** [comment]
LGTM, @nam-jaehyun can you please squash the commits?

**Review by nyrahul — COMMENTED** [review]
LGTM. Lets squash the commits.

**Review by nyrahul — APPROVED** [review]
LGTM. Thanks

**Review by Ankurk99 — APPROVED** [review]

**Pull request merged**